File: src/cpu/file_io/read_output.jl
function read_output(file_path::String)
	file_stream::IOStream = open(file_path, "r")
	file_string::String = readstring(file_stream)
	close(file_stream)

	model::String = read_key(file_string, "model", String)
	number_of_components::Int64 = read_key(file_string, "number_of_components", Int64)
	number_of_abc_samples::Int64 = read_key(file_string, "number_of_abc_samples", Int64)
	m::Array{Float64, 2} = reshape(read_key(file_string, "m", Array{Float64, 1}), (number_of_components, number_of_abc_samples))
	c::Array{Float64, 2} = reshape(read_key(file_string, "c", Array{Float64, 1}), (number_of_components, number_of_abc_samples))
	az::Array{Float64, 2} = reshape(read_key(file_string, "az", Array{Float64, 1}), (number_of_components, number_of_abc_samples))
	dist::Array{Float64, 1} = read_key(file_string, "dist", Array{Float64, 1})
	w::Array{Float64, 1} = read_key(file_string, "w", Array{Float64, 1})
	weighting_scheme::String = read_key(file_string, "weighting_scheme", String)
	gamma::Float64 = read_key(file_string, "gamma", Float64)
	t_exec::Float64 = read_key(file_string, "t_exec", Float64)
	number_of_iterations::Int64 = read_key(file_string, "number_of_iterations", Int64)
	number_of_simulations::Int64 = read_key(file_string, "number_of_simulations", Int64)

	return (
		model,
		number_of_components,
		number_of_abc_samples,
		m,
		c,
		az,
		dist,
		w,
		weighting_scheme,
		gamma,
		t_exec,
		number_of_iterations,
		number_of_simulations)
end

File: matlab_tools/read_output.m
function [  model, ...
            number_of_components, ...
            number_of_abc_samples, ...
            m, ...
            c, ...
            az, ...
            dist, ...
            w, ...
            weighting_scheme, ...
            gamma, ...
            t_exec, ...
            number_of_iterations, ...
            number_of_simulations] = read_output(file_path)

    file_string = fileread(file_path);

    model = read_key(file_string, 'model', 'string');
	number_of_components = read_key(file_string, 'number_of_components', 'scalar');
	number_of_abc_samples = read_key(file_string, 'number_of_abc_samples', 'scalar');
	m = reshape(read_key(file_string, 'm', 'array'), [number_of_components, number_of_abc_samples]);
	c = reshape(read_key(file_string, 'c', 'array'), [number_of_components, number_of_abc_samples]);
    
    az = read_key(file_string, 'az', 'array');
    if numel(az) == number_of_components * number_of_abc_samples
        az = reshape(az, [number_of_components, number_of_abc_samples]);
    else
        az = az(:);
        az = repmat(az, [3 1]);
        az = reshape(az, [number_of_components, number_of_abc_samples]);
    end
	
    dist = read_key(file_string, 'dist', 'array');
	w = read_key(file_string, 'w', 'array');
	weighting_scheme = read_key(file_string, 'weighting_scheme', 'string');
	gamma = read_key(file_string, 'gamma', 'scalar');
	t_exec = read_key(file_string, 't_exec', 'scalar');
	number_of_iterations = read_key(file_string, 'number_of_iterations', 'scalar');
	number_of_simulations = read_key(file_string, 'number_of_simulations', 'scalar');
    
end